File: content/snippets/typescript/interview_lcci/05/01_insert.ts
function insert(N: number, M: number, i: number, j: number): number {
  const mask = ~(((1 << (j - i + 1)) - 1) << i);
  return (N & mask) | (M << i);
}

console.log(insert(1024, 19, 2, 6));
console.log(insert(0, 19, 0, 4));

File: content/snippets/python/interview_lcci/05/01_insert.py
def insert(N: int, M: int, i: int, j: int) -> int:
    mask = ~(((1 << (j - i + 1)) - 1) << i)
    return (N & mask) | (M << i)


def main() -> None:
    print(insert(1024, 19, 2, 6))
    print(insert(0, 19, 0, 4))


if __name__ == "__main__":
    main()
